### ISSUE TITLE
Reviving Telnet access on Xiaomi Gateway 3 ZNDMWG04LM after getting key using UART Method.

### ISSUE THREAD

**Issue by jksmurf** [AlexxIT/XiaomiGateway3#1465]
I have a ZNDMWG04LM with 1.07_0019 FW. 

I did the UART connection and checked the output and succesfully got the Key (not the Token), based on [methods detailed in 1166]. This allowed me to ADD it to AlexxIT's Gateway 3 interface. In fact I did two of them. (https://github.com/AlexxIT/XiaomiGateway3/issues/1166).

My question however is about the need for telnet following integration; I can access the Gateway menu using Putty, but no Admin/Password pair matches, so I cannot actually log in. I tried admin/admin, admin/blank, admin/123123 and a few other combinations. I tried the 5-2-2-2-2-2-2 setting.

- Do I need telnet access for anything in future?
- If yes, is the only way to use UART, then downgrade the FW; and then strip out the password, so if I upgrade it again, it is blank? Is this possible and will blank password this be "sticky" if I upgrade the FW again?
- Is there actually some way to get Full Telnet Access using the Token+Key (which I now have saved) for each device?

Thx
![IMG_E0823](https://github.com/user-attachments/assets/9dc6c448-2a2d-498e-a9b4-3b83613ec39e)

**Comment by AlexxIT** [comment]
This model has root user with blank password (after adding to integration).

**Comment by jksmurf** [comment]
Yes, I can get to the # prompt but it doesn't look like a screen you normally see when you logon using putty.

![Root Blank](https://github.com/user-attachments/assets/a004d4ef-cafa-42f0-87b3-7190d1b0769b)

**Comment by AlexxIT** [comment]
You're in an empty folder. That's why ls doesn't show anything.

**Comment by jksmurf** [comment]
Thank you. Is there a way to WinSCP into these Gateways, I just find it easier to use WinSCP to check folders/files, but I keep getting connection refused?

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/XiaomiGateway3/blob/0acd959050bc221225c7377c3113f4976cd482d3/custom_components/xiaomi_gateway3/core/shell/shell_e1.py#L71-L72

**Comment by jksmurf** [comment]
> https://github.com/AlexxIT/XiaomiGateway3/blob/0acd959050bc221225c7377c3113f4976cd482d3/custom_components/xiaomi_gateway3/core/shell/shell_e1.py#L71-L72

Thank you AlexxIT, but I’m sorry that’s just gone way over my head. I’ve no idea what to do with that 🙏

**Comment by AlexxIT** [comment]
This is command for run FTP

**Comment by jksmurf** [comment]
Thanks @AlexxIT. 

I did a bit of searching around based on that information and got it to work, so for the benefit of others wanting to use WinSCP to poke around in the Gateway File System (own risk!), I jotted down a few steps:

1. After you have removed the password using the UART approach, then login to the Gateway using Putty, username 'root' and no password (just hit enter);

2. After you login using Putty, IF you WANT to add a password back (you do not have to!), issuing this (where #!/bin/sh is the prompt in Putty, you might just see a '#' per the pics and not '#!/bin/sh'), will change the password to "xxxxxx"; remember the new password!

```
#!/bin/sh
echo "root:xxxxxx" | chpasswd
```

Note: If you add a password (please remember it!) and want to revert back to NO password, after you login (with your current password), just run:

```
#!/bin/sh
echo "root:" | chpasswd
```

Note that if you do this, Putty will now give you a password prompt, just hit <Enter> to login i.e. No Password.

3. After you login via Putty, issuing this (where #!/bin/sh is the prompt in Putty), will give you WinSCP access via FTP:

```
#!/bin/sh
/usr/sbin/telnetd
/usr/bin/tcpsvd -E 0.0.0.0 21 ftpd -w / &
```

I believe you only need to issue it once (maybe unless you reboot the device).

4. After Step 3, go to WinSCP, make a new profile using the FTP protocol, using your IP Address, Port 21, user root and (no password unless you added one); no encryption, you're in!

This was my experience, the brief writeup is for users at my level of expertise (low!). There may be other or more correct ways to do these things and I would be happy to have someone add or correct me, no offense.

HTH

k.

![1-Error message without setting up FTP](https://github.com/user-attachments/assets/a669d6d3-1cce-463c-9044-f28dba651f49)
![2-Putty_to_XiaomiGateway](https://github.com/user-attachments/assets/04cec327-d6b8-4294-80c3-3600b0f401e6)
![3-Set WinSCP Access](https://github.com/user-attachments/assets/9f983e3d-d81b-4a2b-9fc3-e72a042ecb37)
![4-Gateway on which I changed the PWD then Back to Blank](https://github.com/user-attachments/assets/494c1015-5a9b-47ba-8ac9-2af9fc9bfc93)
![5-Xiaomi GatewayWiNSCP_Profile](https://github.com/user-attachments/assets/fc3e65b3-e57d-4792-af45-70374a2381a6)
![6-WinSCP Access You are In](https://github.com/user-attachments/assets/9d82a922-5598-437c-8b3e-102294c03a62)
